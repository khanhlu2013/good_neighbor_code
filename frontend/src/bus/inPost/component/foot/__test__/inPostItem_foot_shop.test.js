import React from "react";
import { shallow } from "enzyme";

import InPostItemFootShop from "../inPostItem_foot_shop";
import LoadingIcon from "../../../../../share/loadingIcon";

describe("inPostItem_foot_shop", () => {
  describe("loading icon", () => {
    it("show during requesting post", () => {
      const postId = "1";
      const onRequestPost = jest.fn();

      const wrap = shallow(
        <InPostItemFootShop
          postId={postId}
          isRequestingPost={true}
          onRequestPost={onRequestPost}
        />
      );
      expect(wrap).toMatchSnapshot();
      expect(wrap.find(LoadingIcon)).toHaveLength(1);
      expect(wrap.find("button")).toHaveLength(0);
    });

    it("not show when not requesting post", () => {
      const postId = "1";
      const onRequestPost = jest.fn();

      const wrap = shallow(
        <InPostItemFootShop
          postId={postId}
          isRequestingPost={false}
          onRequestPost={onRequestPost}
        />
      );
      expect(wrap).toMatchSnapshot();
      expect(wrap.find(LoadingIcon)).toHaveLength(0);
      expect(wrap.find("button")).toHaveLength(1);
    });
  });

  it("trigger onRequestPost callback when request btn is clicked", () => {
    const postId = "1";
    const onRequestPost = jest.fn();

    const wrap = shallow(
      <InPostItemFootShop
        postId={postId}
        isRequestingPost={false}
        onRequestPost={onRequestPost}
      />
    );
    wrap.find("#outPostItem-requestBtn-react").simulate("click");
    expect(onRequestPost.mock.calls.length).toBe(1);
    expect(onRequestPost.mock.calls[0][0]).toBe("1");
  });
});
