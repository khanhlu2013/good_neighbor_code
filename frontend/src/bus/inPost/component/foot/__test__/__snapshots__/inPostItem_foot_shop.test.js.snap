// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inPostItem_foot_shop render correctly when isRequestingPost is false 1`] = `
<button
  className="btn btn-success"
  id="outPostItem-requestBtn-react"
  onClick={[Function]}
>
  request
</button>
`;

exports[`inPostItem_foot_shop render correctly when isRequestingPost is true 1`] = `
<LoadingIcon
  text="requesting"
/>
`;
